### PR TITLE
Dynamic Concurrency Requirement  High Level changes required for it

### PR DESCRIPTION
**Pull request by SrinivasChilveri**
What this PR does / why we need it:
This PR explains Dynamic concurrency feature related information and 
major changes required for the same.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
file name `DynamicConcurrency.md` keep consistent with others, use lowercase-lowercase format

File: content/docs/design/DynamicConcurrency.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+toc = true
+aliases = ["/docs/design/"]
+[menu.main]
+  parent = "Design"
+  weight = 2 
++++
+
+# **Dynamic Concurrency feature in KubeGene**
+
+In the process of gene data processing, it is often necessary to read the contents of a file to control concurrent tasks, or to obtain the "output results" of another tasks.
```
Comment (hzxuzhonghu) [inline content/docs/design/DynamicConcurrency.md]: How about change `gene data processing` to a professional phrase like `genome sequencing`

File: content/docs/design/DynamicConcurrency.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+Note that there is only one member in the array. That is, there is only one concurrency.
+If we want, each one as a concurrency, you can add a "segment" to split . See the get_result function for the specific syntax .
+
+    Vars_iter:
+      - get_result( job-1, " ")   
+Will get:
+
+    Vars_iter:
+      - ["1", "2", "3", "4"] 
+This gives four concurrency, each variable being "1", "2", "3", "4". Among them, the separator is an arbitrary string.
```
Comment (hzxuzhonghu) [inline content/docs/design/DynamicConcurrency.md]: plural？

File: content/docs/design/DynamicConcurrency.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      - [A, B, C]
+      - ["list-1.txt", "list-2.txt", "list-3.txt", "list-4.txt"] # <==== Note here, the concurrency of ${2} Quantity (range)
+```
+
+## High Level changes required for this feature
+
+
+* Validations,instantiations etc for get_result  function in genectl 
+
+* changes in the execution crd to include the get_result related job like add command_iter in the task
```
Comment (hzxuzhonghu) [inline content/docs/design/DynamicConcurrency.md]:
```suggestion
* changes in the execution API to include the get_result related job like add command_iter in the task
```

File: content/docs/design/DynamicConcurrency.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+``` go
+// update the Task as below 
+type Task struct {
+	
+	// CommandsIter defines batch command for workflows job.
+	CommandsIter CommandsIter `json:"commands_iter,omitempty"`
+	
+	}
```
Comment (hzxuzhonghu) [inline content/docs/design/DynamicConcurrency.md]: no indent

File: content/docs/design/DynamicConcurrency.md (hunk 1 above)
Comment (SrinivasChilveri) [reply content/docs/design/DynamicConcurrency.md]: changed

File: content/docs/design/DynamicConcurrency.md (hunk 2 above)
Comment (SrinivasChilveri) [reply content/docs/design/DynamicConcurrency.md]: done

File: content/docs/design/DynamicConcurrency.md (hunk 3 above)
Comment (SrinivasChilveri) [reply content/docs/design/DynamicConcurrency.md]: done

File: content/docs/design/DynamicConcurrency.md (hunk 4 above)
Comment (SrinivasChilveri) [reply content/docs/design/DynamicConcurrency.md]: changed

**Pull request merged**